src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,27 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return ((width == null && rectangle.width == null) || width.equals(rectangle.width))
+                && ((length == null && rectangle.width == null) || length.equals(rectangle.length))
+                && ((color == null && color == null) || color.equals(rectangle.color));
```

Comment:
what if `width` is null and `rectangle.width` is not null?

also, if both of them are null, the following code will return true `width == rectangle.width`